cost-analyzer/charts/thanos/templates/query-ingress.yml

```diff
@@ -80,10 +112,20 @@ spec:
     - host: {{ . }}
       http:
         paths:
+          {{- if $apiV1 }}
+          - path: {{ $.Values.query.grpc.ingress.path }}
+            pathType: {{ $.Values.query.grpc.ingress.pathType }}
+            backend:
+              service:
+                name: {{ include "thanos.componentname" (list $ "query") }}-grpc
+                port:
+                  number: {{ $.Values.query.grpc.port }}
+          {{- else }}
           - path: {{ $.Values.query.grpc.ingress.path }}
             backend:
               serviceName: {{ include "thanos.componentname" (list $ "query") }}-grpc
-              servicePort: {{ $.Values.query.http.port }}
+              servicePort: {{ $.Values.query.grpc.port }}
```

Comment:
I think this is probably a great bug fix, just want to check w/ @AjayTripathy in case it was intentional

Reply:
Uhmm you can't look at the UI via the GRPC port, so I'm not sure this was intentional. @gracedo what was the thinking here?

Reply:
Referencing the upstream thanos helm chart, this ingress also exists https://github.com/bitnami/charts/blob/main/bitnami/thanos/templates/query/ingress-grpc.yaml likely because some use cases require being able to access the gRPC service externally (e.g. hierarchical thanos where a central querier may need to connect to various queriers running on different cluster). I imagine this ingress was originally carried over from the upstream chart when bundled into the subchart here